crates/stackable-versioned-macros/src/utils/mod.rs

```diff
@@ -58,82 +56,30 @@ impl ContainerIdentExt for IdentString {
     }
 }
 
-pub trait ItemIdentExt: Deref<Target = IdentString> + From<Ident> + Spanned {
-    const DEPRECATED_PREFIX: &'static str;
-
-    fn deprecated_prefix(&self) -> &'static str {
-        Self::DEPRECATED_PREFIX
-    }
-
-    fn starts_with_deprecated_prefix(&self) -> bool {
-        self.deref().as_str().starts_with(Self::DEPRECATED_PREFIX)
-    }
-
-    /// Removes deprecation prefixed from field or variant idents.
-    fn as_cleaned_ident(&self) -> IdentString;
-}
-
-pub struct FieldIdent(IdentString);
-
-impl ItemIdentExt for FieldIdent {
-    const DEPRECATED_PREFIX: &'static str = "deprecated_";
-
-    fn as_cleaned_ident(&self) -> IdentString {
-        self.0
-            .clone()
-            .map(|i| i.trim_start_matches(Self::DEPRECATED_PREFIX).to_string())
-    }
-}
-
-impl From<Ident> for FieldIdent {
-    fn from(value: Ident) -> Self {
-        Self(IdentString::from(value))
-    }
-}
+pub trait ItemIdents {
+    const DEPRECATION_PREFIX: &str;
 
-impl Deref for FieldIdent {
-    type Target = IdentString;
-
-    fn deref(&self) -> &Self::Target {
-        &self.0
+    fn deprecation_prefix(&self) -> &str {
+        Self::DEPRECATION_PREFIX
     }
-}
 
-impl ToTokens for FieldIdent {
-    fn to_tokens(&self, tokens: &mut proc_macro2::TokenStream) {
-        self.0.to_tokens(tokens);
+    fn starts_with_deprecation_prefix(&self) -> bool {
+        self.original()
+            .as_str()
+            .starts_with(Self::DEPRECATION_PREFIX)
     }
-}
-
-pub struct VariantIdent(IdentString);
-
-impl ItemIdentExt for VariantIdent {
-    const DEPRECATED_PREFIX: &'static str = "Deprecated";
 
-    fn as_cleaned_ident(&self) -> IdentString {
-        self.0
-            .clone()
-            .map(|i| i.trim_start_matches(Self::DEPRECATED_PREFIX).to_string())
-    }
-}
-
-impl From<Ident> for VariantIdent {
-    fn from(value: Ident) -> Self {
-        Self(IdentString::from(value))
-    }
+    fn cleaned(&self) -> &IdentString;
+    fn original(&self) -> &IdentString;
 }
 
-impl Deref for VariantIdent {
-    type Target = IdentString;
-
-    fn deref(&self) -> &Self::Target {
-        &self.0
-    }
+pub trait ItemIdentExt {
+    fn json_path_ident(&self) -> IdentString;
 }
 
-impl ToTokens for VariantIdent {
-    fn to_tokens(&self, tokens: &mut proc_macro2::TokenStream) {
-        self.0.to_tokens(tokens);
+impl ItemIdentExt for IdentString {
+    fn json_path_ident(&self) -> IdentString {
+        format_ident!("__sv_{}_path", self.as_str().to_lowercase()).into()
```

Comment:
I think it's nicer to give the interpolation a name to lower cognitive load

```suggestion
        format_ident!("__sv_{ident_name}_path", ident_name = self.as_str().to_lowercase()).into()
```

or similar (`cargo fmt` would probably change that in some way).

Reply:
Oh yeah, that makes sense. I will update it.